### PR TITLE
Feat/aws

### PR DESCRIPTION
**Pull request by rossman-andrew**
https://trello.com/c/nuCdeUv5/77-generate-output-for-message-bus-on-completed-user-session
https://trello.com/c/tQNYuyo1/98-set-up-amazon-sqs-for-events-service
https://trello.com/c/8wmWEUrg/87-refactor-and-debug-movie-progress-for-user-events
https://trello.com/c/E8smtPeA/49-refactor-data-generation-to-simulate-reasonable-order-of-user-watching-behavior

### CODE REVIEW

File: database/dummyData/userEventData.js [hunk 1]
```diff
@@ -102,69 +103,82 @@ const generateMovieObj = () => {
...
+    time: t,
   };
   return query;
 };
 
+const periodStart = new Date('August 5, 2017 00:00:01').getTime();
+const periodEnd = new Date('November 5, 2017 23:59:59').getTime();
+const periodDuration = periodEnd - periodStart;
+
+const randTime = () => periodStart + (Math.random() * periodDuration);
```
Comment (shwetachari) [inline database/dummyData/userEventData.js:120]: Interesting timestamp generation! 👍

File: database/dummyData/userEventData.js [hunk 2]
```diff
@@ -214,97 +214,170 @@ const simulateUserEvents = (numOfSessions) => {
...
-    eventDashboard = [];
+    eventArray = simulateUserEvents(1);
+    index = -1;
   }
-};
+});
+
+// **** FOR SEED DATA GENERATION *****
+
+// const elasticInsert = () => {
```
Comment (shwetachari) [inline database/dummyData/userEventData.js]: Commenting out this part makes the code look somewhat unmeaningful/purposeless. Maybe you could just comment out the part that actually runs this code (i.e. the cron worker)?

File: server/index.js [hunk 3]
```diff
@@ -41,6 +58,123 @@ app.post('/userEventData', (req, res) => {
...
+        })
+        : prev;
+    }, []),
+  };
+};
+
+let eventDashboard = [];
+app.post('/newEvent', (req, res) => {
+  db.addEvent(req.body.session, req.body.query)
+    .then((msgObj) => {
```
Comment (claracho) [inline server/index.js:113]: Lines 113-142 can all be inside one then() body, what is the reason behind chaining them?

**Review by shwetachari — APPROVED** [review]
I only have one major point of concern, which should be simple to address. Other than that, looks good!

File: server/index.js [hunk 4]
```diff
@@ -41,6 +58,123 @@ app.post('/userEventData', (req, res) => {
...
+      }
+      return data;
+    })
+    .then((data) => {
+      if (data) {
+        res.sendStatus(201);
+      }
+      return null;
+    })
+    .catch((err) => {
```
Comment (claracho) [inline server/index.js:155]: This catch block is not capturing errors that may get generated by AWS SQS/dashboard functions.

**Review by claracho — COMMENTED** [review]
The promise chain for /newEvent can be modified to better handle/capture errors that maybe generated during asynchronous operations.

**Pull request merged**